src/instrumentation_measurement/measurements.py

```diff
@@ -159,6 +165,45 @@ def csv_row_append_info(self, instrumenter_result, program, sample_id):
     to_append = "{},{},".format(program.id, sample_id)
     return [to_append + row for row in instrumenter_result]
 
+  def sanitize_tracer_result(self, tracer_result):
+      specs = self.read_opcodes_specs()
+
+      result = []
+      for row in tracer_result:
+        row = row.split(',')
+        prefix = row[0:4] # take first columns untouched
+        opcode = row[2]
+        stack_depth = int(row[3])
+        stack = row[4:]
+
+        # stack top is stack[stack_depth - 1] and stack bottom is stack[0],
+        # so to take some deeper arguments we have to do some mysterious maths
+        args = [stack[stack_depth - arg_depth - 1] for arg_depth in specs[opcode]]
+
+        # append with empty positions to conform to CSV format
+        for i in range(len(args), MAX_OPCODE_ARGS):
+            args.append('')
+
+        parsed_line = ','.join(prefix + args)
+        result.append(parsed_line)
+
+      return result
+
+  def read_opcodes_specs(self):
+      with open('./instrumentation_measurement/data/opcodes_args.csv', newline='') as opcodes_args_file:
```

Comment:
probably safer to use path relevant to this file, see `program_generator.py` for example

Reply:
2nd comment: can't we just derive the information we need from the `opcodes.csv` file we have? Arity of the OPCODEs should be enough, except SWAP and DUP.

I'm afraid these files would be cumbersome to keep in sync

Reply:
I guess that it might be a good idea - I can merge it :)